ykrt/src/mt.rs

```diff
@@ -41,14 +45,62 @@ const PHASE_COUNTING: usize = 0b10; // The value specifies how many times we've
 /// mechanism is for every bytecode or AST node to have its own `Location` (even for bytecodes or
 /// nodes that can't be control points).
 #[derive(Debug)]
-pub struct Location {
-    pack: AtomicUsize,
+pub struct Location<I> {
+    // A Location is a state machine which operates as follows (where PHASE_COUNTING is the start
+    // state):
+    //
+    //               ┌────────────────────────────────────────┐
+    //               │                                        │ ─────────────┐
+    //   reprofile   │             PHASE_COUNTING             │              │
```

Comment:
We don't implement that reprofile stage yet, do we?

Reply:
Not yet, no.